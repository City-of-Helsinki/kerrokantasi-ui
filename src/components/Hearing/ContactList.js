import React from 'react';
import PropTypes from 'prop-types';
import {Row, Col} from 'react-bootstrap';
import ContactCard from '../ContactCard';
import {FormattedMessage} from 'react-intl';

export const ContactListComponent = ({contacts, language}) => {
  return (
    <div className="hearing-contacts">
      <h3>
        <FormattedMessage id="contactPersons" />
      </h3>
      <Row>
        {contacts &&
          contacts.map((person, index) => (
            <Col
              xs={6}
              key={index + Math.random()} // eslint-disable-line react/no-array-index-key
              md={4}
            >
              <ContactCard activeLanguage={language} {...person} />
            </Col>
          ))}
      </Row>
    </div>
  );
};

ContactListComponent.propTypes = {
  contacts: PropTypes.array,
  language: PropTypes.string
};

export default ContactListComponent;
