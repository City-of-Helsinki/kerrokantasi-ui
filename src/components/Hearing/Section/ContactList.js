import React from 'react';
import PropTypes from 'prop-types';
import {Row} from 'react-bootstrap';
import ContactCard from '../../ContactCard';

export const ContactListComponent = ({contacts, language}) => {
  return (
    <Row>
      {contacts && contacts.map((person, index) => (
        <ContactCard
          activeLanguage={language}
          key={index + Math.random() /* eslint-disable-line react/no-array-index-key */}
          {...person}
        />
      ))}
    </Row>
  );
};

ContactListComponent.propTypes = {
  contacts: PropTypes.array,
  language: PropTypes.string
};

export default ContactListComponent;
