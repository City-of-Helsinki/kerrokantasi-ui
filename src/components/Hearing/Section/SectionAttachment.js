import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../../../utils/Icon';


/**
 * Determines the correct title to return
 * @param {Object} title - title object
 * @param {String} language - language of application
 */
const getFileTitle = (title, language) => {
  if (title && title[language] && typeof title[language] === 'string') {
    return title[language];
  }
  return title[Object.keys(title)];
};

/**
 * Class declaration
 * Renders list of attachments available to a section.
 */
const SectionAttachment = (props) => (
  <div className="section-attachment">
    <Icon className="icon" name="file"/>
    <a href={props.file.url} className="section-attachment-title" target="_blank" rel="noreferrer">
      {getFileTitle(props.file.title, props.language)}
    </a>
  </div>
);

SectionAttachment.propTypes = {
  file: PropTypes.object.isRequired,
  language: PropTypes.string.isRequired,
};

export default SectionAttachment;
export { getFileTitle };
