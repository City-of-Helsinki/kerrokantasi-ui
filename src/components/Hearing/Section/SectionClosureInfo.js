/* eslint-disable react/no-danger */
import React from 'react';
import PropTypes from 'prop-types';

export const SectionClosureInfoComponent = ({content}) => {
  return (
    <div className="closure-info">
      <div className="container">
        <div dangerouslySetInnerHTML={{ __html: content }} />
      </div>
    </div>
  );
};

SectionClosureInfoComponent.propTypes = {
  content: PropTypes.string
};

export default SectionClosureInfoComponent;
