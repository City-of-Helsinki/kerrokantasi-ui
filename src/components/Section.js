/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import SortableCommentList from './SortableCommentList';
import {Link} from 'react-router';
import Icon from '../utils/Icon';
import {isSpecialSectionType, userCanComment} from '../utils/section';
import classNames from 'classnames';
import MapdonHKRPlugin from './plugins/legacy/mapdon-hkr';
import MapdonKSVPlugin from './plugins/legacy/mapdon-ksv';
import MapQuestionnaire from './plugins/MapQuestionnaire';
import Alert from 'react-bootstrap/lib/Alert';
import getAttr from '../utils/getAttr';

function getImageList(section, language) {
  if (section.type === "main") { // Main section images aren't rendered here atleast atm.
    return null;
  }
  return section.images.map((image) => (<div key={image.url}>
    <img
      className="img-responsive"
      alt={getAttr(image.title, language)}
      title={getAttr(image.title, language)}
      src={image.url}
    />
    <div className="image-caption">{getAttr(image.caption, language)}</div>
  </div>));
}


export default class Section extends React.Component {
  constructor(props) {
    super(props);
    this.state = {collapsed: true};
  }

  // componentDidMount() {
  //   // if (!this.isCollapsible()) {  // Trigger immediate comment load for uncollapsible sections
  //   //   this.loadComments();
  //   // }
  // }

  componentWillUpdate(nextProps, nextState) {
    if (this.state.collapsed && !nextState.collapsed) {  // (Re-)load comments when uncollapsing
      // this.loadComments();
    }
  }

  onPostComment(text, authorName, pluginData, geojson, label, images) {
    const {section} = this.props;
    const commentData = {text, authorName, pluginData, geojson, label, images};
    if (this.props.onPostComment) {
      this.props.onPostComment(section.id, commentData);
    }
  }

  onPostVote(commentId) {
    const {section} = this.props;
    this.props.onPostVote(commentId, section.id);
  }

  toggle() {
    this.setState({collapsed: !this.state.collapsed});
  }

  loadComments() {
    const {section} = this.props;
    if (this.props.loadSectionComments) {
      this.props.loadSectionComments(section.id);
    }
  }

  getTitleDiv(collapsed, collapsible) {
    const {section, linkTo} = this.props;
    const {language} = this.context;
    if (section.type === "main") {
      return null;
    }
    if (section.type === "closure-info") {
      return (
        <h3 className="section-title">
          {getAttr(section.title, language)}
        </h3>
      );
    }
    const iconName = (collapsed ? "chevron-right" : "chevron-down");

    if (linkTo) {
      return (
        <h2 className="section-title" onClick={this.toggle.bind(this)}>
          <Link to={linkTo}>
            {collapsible ? (<span><Icon name={iconName}/>&nbsp;</span>) : null}
            {getAttr(this.props.section.title, language)}
          </Link>
          {collapsed ? (
            <div className="section-title-comments">
              <Icon name="comment-o"/>&nbsp;{section.n_comments}
            </div>
          ) : null}
        </h2>
      );
    }

    return (
      <h2 className="section-title" onClick={this.toggle.bind(this)}>
        {collapsible ? (<span><Icon name={iconName}/>&nbsp;</span>) : null}
        {getAttr(this.props.section.title, language)}
        {collapsed ? (
          <div className="section-title-comments">
            <Icon name="comment-o"/>&nbsp;{section.n_comments}
          </div>
        ) : null}
      </h2>
    );
  }

  renderPluginContent(section) {
    if (!this.props.showPlugin) {
      return null;
    }
    const {user} = this.props;
    const comments = this.props.comments ? this.props.comments.data : [];
    if (typeof window === 'undefined' || !section.plugin_identifier) {
      return null;
    }
    switch (section.plugin_identifier) {
      case "mapdon-hkr":
        return (
          <MapdonHKRPlugin
            data={section.plugin_data}
            onPostComment={this.onPostComment.bind(this)}
          />
        );
      case "mapdon-ksv":
        return (
          <MapdonKSVPlugin
            data={section.plugin_data}
            onPostComment={this.onPostComment.bind(this)}
            pluginPurpose="postComments"
            canSetNickname={user === null}
          />
        );
      case "map-questionnaire":
        return (
          <MapQuestionnaire
            data={section.plugin_data}
            onPostComment={this.onPostComment.bind(this)}
            onPostVote={this.onPostVote.bind(this)}
            comments={comments}
            pluginPurpose="postComments"
            canSetNickname={user === null}
            displayCommentBox={false}
            pluginSource={section.plugin_iframe_url}
          />
        );
      default:
        return <Alert>I do not know how to render the plugin {section.plugin_identifier}</Alert>;
    }
  }

  isCollapsible() {
    const {section, isCollapsible} = this.props;
    const hasPlugin = !!section.plugin_identifier;
    return isCollapsible && !isSpecialSectionType(section.type) && !hasPlugin;
  }

  isCommentable() {
    const {section} = this.props;
    const hasPlugin = !!section.plugin_identifier;
    return (this.props.canComment && !hasPlugin);
  }

  render() {
    const {section, user} = this.props;
    const {language} = this.context;
    const collapsible = this.isCollapsible();
    const collapsed = collapsible && this.state.collapsed;
    const titleDiv = this.getTitleDiv(collapsed, collapsible);
    let commentList = null;
    if (collapsed) {
      return (
        <div className="section-list-item">
          <div className="section-list-item-image" onClick={this.toggle.bind(this)}>
            {section.images.length ? <img alt="" src={section.images[0].url} /> : null}
          </div>
          <div className="section-list-item-content">
            {titleDiv}
            <div className="section-abstract" dangerouslySetInnerHTML={{__html: getAttr(section.abstract, language)}} />
          </div>
          <hr/>
        </div>
      );
    }

    if (!isSpecialSectionType(section.type)) {
      commentList = (<SortableCommentList
        section={section}
        comments={(this.props.comments ? this.props.comments.results : null) || []}
        canComment={this.isCommentable() && userCanComment(this.props.user, section)}
        onPostComment={this.onPostComment.bind(this)}
        canVote={this.props.canVote}
        onPostVote={this.onPostVote.bind(this)}
        canSetNickname={!user}
        isSectionComments={section}
        onDeleteComment={this.props.handleDeleteClick}
        onEditComment={this.props.onEditComment}
      />);
    }
    const imageList = getImageList(section, language);
    const sectionClass = classNames({
      'hearing-section': true,
      'closure-info': section.type === "closure-info"
    });
    const pluginContent = this.renderPluginContent(section);
    return (<div className={sectionClass}>
      {titleDiv}
      <div className="section-content">
        {imageList}
        {section.type !== "main" ?
          <div
            className="section-abstract lead"
            dangerouslySetInnerHTML={{__html: getAttr(section.abstract, language)}}
          /> :
          null}
        <div dangerouslySetInnerHTML={{__html: getAttr(section.content, language)}} />
        {pluginContent}
      </div>
      {commentList}
      <hr/>
    </div>);
  }
}

Section.defaultProps = {
  showPlugin: true,
  isCollapsible: true,
};

Section.propTypes = {
  canComment: React.PropTypes.bool,
  canVote: React.PropTypes.bool,
  comments: React.PropTypes.object,
  isCollapsible: React.PropTypes.bool,
  linkTo: React.PropTypes.oneOfType([
    React.PropTypes.string,
    React.PropTypes.object,
    React.PropTypes.func
  ]),
  loadSectionComments: React.PropTypes.func,
  onPostComment: React.PropTypes.func,
  onPostVote: React.PropTypes.func,
  section: React.PropTypes.object.isRequired,
  showPlugin: React.PropTypes.bool,
  user: React.PropTypes.object,
  handleDeleteClick: React.PropTypes.func,
  onEditComment: React.PropTypes.func,
};

Section.contextTypes = {
  language: React.PropTypes.string.isRequired
};
