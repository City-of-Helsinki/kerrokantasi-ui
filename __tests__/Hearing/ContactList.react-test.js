import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {ContactListComponent} from '../../src/components/Hearing/Section/ContactList';
import {mockStore, getIntlAsProp} from '../../test-utils';

// Renders the Hearings component using enzymes shallow rendering.
// You can pass props you want to override as a parameter.
const setup = propOverrides => {
  const {mockHearingWithSections} = mockStore;
  const props = Object.assign({
    contacts: mockHearingWithSections.data.contacts,
    language: 'fi'
  }, propOverrides);

  const wrapper = shallow(<ContactListComponent {...props} />);

  return {
    props,
    wrapper
  };
};

test('ContactList component should render as expected', () => {
  const {wrapper} = setup();
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
