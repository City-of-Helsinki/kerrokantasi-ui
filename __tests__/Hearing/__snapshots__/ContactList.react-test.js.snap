// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList component should render as expected 1`] = `
<div
  className="hearing-contacts"
>
  <h3>
    <FormattedMessage
      id="contactPersons"
      tagName="span"
      values={Object {}}
    />
  </h3>
  <Row
    bsClass="row"
    componentClass="div"
  />
</div>
`;

exports[`Header component should render as expected 1`] = `
<div
  className="hearing-contacts"
>
  <h3>
    <FormattedMessage
      id="contactPersons"
      tagName="span"
      values={Object {}}
    />
  </h3>
  <Row
    bsClass="row"
    componentClass="div"
  />
</div>
`;
